### PR TITLE
Minor updates to Scorer

### PR DESCRIPTION
**Pull request by dwhswenson**
This is a small PR where I'm cherry-picking some of the changes that have come up as I've been working on AtomMappers. Main things:

* Added `setup/errors.py`, which currently just contains a string template for errors when using abstract methods.
* Make `score` and `annotation` functions in `Scorer` private. This will allow arbitrary functions to be used as scorers (useful if an AtomMapper wants to provide a scoring function as an instance method).
* Move the fixtures for atommappings into conftest.py; use one of them instead of the nonsense placeholder for the scoring tests.

### CODE REVIEW

**Review by richardjgowers — APPROVED** [review]
LGTM

File: openfe/setup/errors.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+ABSTRACT_ERROR_STRING = ("'{cls}' is an abstract class and should not be "
```
Comment (IAlibay) [inline openfe/setup/errors.py:1]: Do we envision having abstract classes outside of setup? If so maybe it needs to be in the top level directory?

**Review by IAlibay — APPROVED** [review]
lgtm just see the one comment re: location of errors.py.

File: openfe/setup/errors.py (hunk 1 above)
Comment (dwhswenson) [reply openfe/setup/errors.py:1]: I had the samne thought. So far, I'm operating under the assumption that we are likely to split into multiple repos (something we should discuss on Monday, probably).

If multiple repos, obviously it must be here. In single repo, then I agree that it belongs top level.

Comment (IAlibay) [reply openfe/setup/errors.py:1]: How about a temporary TODO so we get back to it later? (not sure how folks feel about this, there are fun gh bots that will scrape all the TODO comments and turn them into issues for you)

(feel free to say no and merge tbh, it's not like we're likely to forget either)

```suggestion
# TODO: move if we don't split to multiple repos
ABSTRACT_ERROR_STRING = ("'{cls}' is an abstract class and should not be "
```

Comment (dwhswenson) [reply openfe/setup/errors.py:1]: I agree with the principle of leaving a TODO on this sort of thing, but in this specific case, I'm hoping we'll decide about whether to split repos very soon -- so then it'll be decided soon enough that we don't need the reminder. And then if we need to move the file, it'll be a quick PR right after the meeting! 😊

**Comment by IAlibay** [comment]
huh weird - I merged this, and it seems like the changes went through.. but somehow the PR hasn't closed and now claims conflicts? O.o

**Comment by dwhswenson** [comment]
> huh weird - I merged this, and it seems like the changes went through.. but somehow the PR hasn't closed and now claims conflicts? O.o

That *is* all kinds of weird... I'll try sync'ing the branch locally and see what shakes out

**Comment by dwhswenson** [comment]
So, sync'ing (just `git pull upstream main && git push`, which only pulled in the changes from #27, I think) seems to have automatically moved the status to "closed"... which is kind of annoying, because I would want the status to be "merged"! (purple, not red!)

**Comment by IAlibay** [comment]
yeah :/ that's definitely a new one to me, never seen github bug out like this on a merge.

**Comment by dwhswenson** [comment]
Just curious -- regular merge or squash merge? I'm not sure how squashing handles cherry-picked commits (although I'd expect any problem to show up in #30, not here).

**Comment by IAlibay** [comment]
Pretty sure it was a regular merge.

**Comment by dwhswenson** [comment]
FWIW, running the [GraphQL Explorer](https://docs.github.com/en/graphql/overview/explorer) shows that a search for merged PRs shows this one.

Query:

```graphql
query {
  repository(name: "openfe", owner: "openfreeenergy") {
    pullRequests(states: MERGED, last: 2) {
      edges {
        node {
          merged
          number
          title
        }
      }
    }
  }
}
```

Results:

```json
{
  "data": {
    "repository": {
      "pullRequests": {
        "edges": [
          {
            "node": {
              "merged": true,
              "number": 27,
              "title": "Prevent CodeCov from running on forks, cron jobs"
            }
          },
          {
            "node": {
              "merged": true,
              "number": 28,
              "title": "atommapping implementation"
            }
          }
        ]
      }
    }
  }
}
```

Anyway, no more debugging from me on that one, though I might search around for similar problems (and might make a post on github.community to see if anyone can explain).

**Comment by dwhswenson** [comment]
Posted issue on github.community: https://github.community/t/pr-marked-as-conflicts-after-merge-syncing-incorrectly-marks-it-closed/226240

**Pull request closed without merging**